fastly/client.go

```diff
@@ -198,7 +203,13 @@ func (c *Client) Request(verb, p string, ro *RequestOptions) (*http.Response, er
 		return nil, err
 	}
 
+	if verb != "GET" && verb != "HEAD" {
```

Comment:
I wonder if it makes sense to use an inclusion list instead of exclusions. This might be missing `PURGE`, for example.

```suggestion
	if verb == "POST" || verb == "PUT" || verb == "PATCH" || verb == "DELETE" {
```

Reply:
I prefer exclusion lists for things like this so it fails safe (more locking) if I've missed something.